lips/lip-4.md

```diff
@@ -0,0 +1,222 @@
+      from_subaddress_not_present, 
+      referenced_event_not_present */
+```
+
+## Submitting a transaction
+
+With the metadata in hand, the sender can now submit a transaction to the Libra block chain via a deposit call in Move:
+
+```
+deposit<LBR>(
+    recipient_address, 
+    amount, 
+    metadata, 
+    metadata_signature);
+```
+
+In the case of the example above, 0x1234 would be the recipient and 20 LBR would be the amount.  Note that metadata_signature must only be present for travel-rule cases between VASPs.
+
+## Processing the transaction
+
+Much like any other funds transfer request, validators only verify that the sender has the sufficient libra to support the transaction and that the transaction is properly formed and valid, they do not inspect or verify the correctness of the metadata.
+
+# Transaction Examples
+
+The following examples demonstrate how subaddressing and metadata are used in the transaction flow.  *Note that the terminology “NC” will mean non-custodial account and “C” will mean a custodial account.*
+
+## NC to NC transaction Flow
+
+For NC to NC transactions, there is no usage of subaddressing/metadata.
+
+## NC to C Transaction Flow
+
+User A (address 0x1234) on a NC wallet wishes to send 100 microlibra to merchant B who is on a private custodial wallet (Where the custodial wallet has a public address of 0x7777 and the merchant has a sub-account of 0x987).  User A's client now composes a raw transaction with the following relevant fields:
+
+```
+metadata = GeneralMetadatav0 {
+  to_subaddress: 0x987,
+};
+
+program = deposit<LBR>(
+    0x7777 /*recipient*/,
+    100 /*amount*/, 
+    lcs(MetadataType::GeneralMetadataType(
+        GeneralMetadata::GeneralMetadataVersion0(metadata))),
+    None /*metadata_signature*/);
+
+RawTransaction {
+    sender_account: 0x1234,
+    program: program, 
+}
+```
+
+## C to NC transaction flow
+
+User A who is on a custodial wallet (Where the C wallet has a public address of 0x7777 and user A has a sub-account of 0x987) wishes to send 100 microlibra to merchant B who is on a NC wallet (with an address of 0x1234).  User A's wallet then composes a transaction via:
+
+```
+metadata = GeneralMetadatav0 {
+  from_subaddress: 0x987,
+};
+
+program = deposit<LBR>(
+    0x1234 /*recipient*/,
+    100 /*amount*/, 
+    lcs(MetadataType::GeneralMetadataType(
+        GeneralMetadata::GeneralMetadataVersion0(metadata))),
+    None /*metadata_signature*/);
+
+RawTransaction {
+    sender_account: 0x7777,
+    program: program, 
+}
+```
+
+
+## Refunds
+
+Merchant B now wishes to refund user A. But user A was sending from a custodial account so merchant B must send the funds back to the custodial account and include subaddress information so that the funds are directed back to user A.  Merchant B’s client now constructs a transaction via the following where referenced_event is the committed event sequence number under the sending account of the original send payment event:
```

Comment:
```suggestion
Merchant B now wishes to refund user A. But user A was sending from a custodial account so merchant B must send the funds back to the custodial account and include subaddress information so that the funds are directed back to user A.  Merchant B’s client now constructs a transaction via the following where referenced_event is the committed event sequence number under the sending account of the original sent payment event:
```